temporal-sdk/src/main/java/io/temporal/workflow/unsafe/WorkflowUnsafe.java

```diff
@@ -66,6 +66,34 @@ public static boolean isReplaying() {
     return WorkflowInternal.isReplaying();
   }
 
+  /**
+   * Runs the supplied procedure in the calling thread with disabled deadlock detection if called
+   * from the workflow thread. Does nothing except the procedure execution if called from a
+   * non-workflow thread.
+   *
+   * @param proc to run with disabled deadlock detection
+   */
+  public static void deadlockDetectorOff(Functions.Proc proc) {
```

Comment:
We chose not to expose these in Go, just left them in internal code read to expose if/when we wanted. Instead we offered the more specific way to create a data converter with deadlock disabled. I wonder if it'd be worth adding a `default DataConverter withWorkflowDeadlockDetectorDisabled()` to the `DataConverter` interface (though I can understand if you don't want that class referencing the workflow package).

But I have no strong opinion.

Reply:
I think users should strive to isolate the "deadlock disabled" segment of code as much as only possible. So, pieces of codec or payload converter that are not performing IO shouldn't go under the disabled deadlock.
I think we shouldn't encourage to just disabling of deadlock detector for the whole DataConverter assembly.
We can raise it on a standup maybe.

Reply:
> I think we shouldn't encourage to just disabling of deadlock detector for the whole DataConverter assembly.

It's a tough tradeoff, because arguably this encourages disabling deadlock detector everywhere.

> Let's sync up on standup.

:+1:

Reply:
> It's a tough tradeoff, because arguably this encourages disabling deadlock detector everywhere.

I'm not exactly worried about it putting it into WorkflowUnsafe. All tools in that class are expected to be used with huge caution and not treated lightly. But good note, I will actually add an explicit doc that these new methods shouldn't be used in actual workflow code implementations.